packages/agreement-process/src/services/agreementContractBuilder.ts

```diff
@@ -127,16 +129,29 @@ const getAttributeInvolved = async (
 };
 
 const getSubmissionInfo = async (
```

Comment:
@MalpenZibo Does this apply only to this `getSubmissionInfo` case or also to the `getActivationInfo` below?
If it applies also to that case, maybe you could move this logic from here into the `retrieveUser`, which is called by both functions

Reply:
it can be applied just to submission info.
The submission can be performed only by the consumer, and we can safely take the details from it.

The activation can be done both by consumer and by producer, but in both cases the `getActivationInfo` is called by the user who also activated the agreement. This is why we can take the info starting from the context (of the jwt)